presentation/matrix_view.py

```diff
@@ -1,153 +1,200 @@
-    annotations = []
-    for row in reversed(mesh.rows):
-        for col in row.columns:
-            text = get_text(metric, matrix.cells[row.agent_alias][col.agent_alias])
-            annotations.append(
-                dict(
-                    showarrow=False,
-                    text=text,
-                    xref="x",
-                    yref="y",
-                    x=col.agent_alias,
-                    y=row.agent_alias,
-                )
-            )
-    return annotations
-
-
-def get_text(metric: str, cell: MeshColumn) -> str:
-    if metric == "latency":
-        return f"<b>{(cell.latency_microsec.value * 1e-3):.2f} ms</b>"
-    elif metric == "jitter":
-        return f"<b>{(cell.jitter_microsec.value * 1e-3):.2f} ms</b>"
-    else:
-        return f"<b>{cell.packet_loss_percent.value:.1f}%</b>"
 
+class MatrixView:
+    MATRIX = "matrix"
+    METRIC_SELECTOR = "metric-selector"
+
+    @staticmethod
+    def make_layout(mesh: MeshResults, metric: Metric) -> html.Div:
+        localtime_timestamp = utc_to_localtime(mesh.utc_timestamp)
+        timestamp = localtime_timestamp.strftime("%H:%M:%S")
+        header = "SLA Dashboard"
+        return html.Div(
+            children=[
+                html.H1(children=header, style={"textAlign": "center", "marginBottom": 50}),
+                html.H2(
+                    f"Data timestamp {timestamp}",
+                    style={"textAlign": "center", "marginTop": 100},
+                ),
+                html.Center("Select primary metric: "),
+                dcc.Dropdown(
+                    id=MatrixView.METRIC_SELECTOR,
+                    options=[
+                        {"label": "Latency [ms]", "value": Metric.LATENCY.value},
+                        {"label": "Jitter [ms]", "value": Metric.JITTER.value},
+                        {"label": "Packet loss [%]", "value": Metric.PACKET_LOSS.value},
+                    ],
+                    value=metric.value,
+                    clearable=False,
+                    style={"width": 450, "margin-left": "auto", "margin-right": "auto"},
+                ),
+                html.Div(
+                    html.Center(
+                        dcc.Graph(id=MatrixView.MATRIX, style={"width": 750, "height": 750}),
+                        style={"marginLeft": 200, "marginRight": 200},
+                    ),
+                ),
+            ],
+            style={"marginBottom": 50, "marginTop": 50, "marginLeft": 50, "marginRight": 50},
+        )
 
-def make_hover_text(mesh: MeshResults, matrix: Matrix) -> List[List[str]]:
-    text = []
-    for row in reversed(mesh.rows):
-        text_col = []
-        for col in row.columns:
+    @staticmethod
+    def make_matrix_data(mesh: MeshResults, metric: Metric, config: Config) -> Dict:
+        matrix = Matrix(mesh)
+        data = MatrixView.make_data(mesh, metric, matrix, MatrixView.get_tresholds(metric, config))
```

Comment:
Why you need to pass the output from `MatrixView.get_tresholds(metric, config)` to `MatrixView.make_data`? Why cannot `MatrixView.make_data` call the same internally?

Is there really any benefit of all methods of the `MatrixView` class being static? Why not have real instances and store `config` and `matrix` in instance variables.

At minimum, methods that need to call other methods of the same class should be @classmethods as opposed to static, so that the name of the class is available to the them. The net effect look like this:

```
@classmethod
def make_matrix_data(cls, mesh: MeshResults, metric: Metric, config: Config) -> Dict:
    matrix = Matrix(mesh)
    data = cls.make_data(mesh, metric, matrix, cls.get_tresholds(metric, config))
    ...
```

Also, I'd say that `make_view_data` would be a better name for this method.

Reply:
This class requires a major refactor, will get this done.